docs/ff-concepts/adding-customization/common-examples.md

```diff
@@ -0,0 +1,284 @@
+String? getProductName(ProductStruct product) {
+    // Access and return the product name
+    return product.name;
+}
+```
+- **Returning a List of Review Comments from ProductStruct**
+
+This function retrieves a list of review comments from the reviews field in the `ProductStruct`. The return type is `List<String>` as it returns a list of comments (or an empty list if there are no reviews).
+
+```js
+// Function to return a list of review comments from a ProductStruct instance
+List<String> getProductReviewComments(ProductStruct product) {
+  // Check if reviews are present and return a list of review comments
+  return product.reviews?.map((review) => review.comment ?? '').toList() ?? [];
+}
+```
+
+#### Example 3: Modifying Properties of an Existing `ProductStruct` Object
+You can also modify the properties of an existing `ProductStruct` object. This can be helpful if you want to update a field before saving the data back to Firebase or passing it into an action.
+
+- **Simple Property Modification**
+In this example, we’ll modify a single property, like `productId`, of an existing `ProductStruct` object. This example is straightforward and demonstrates how to update a basic field in the object.
+
+```js
+// Function to update the product ID of a ProductStruct instance
+Future updateProductId(ProductStruct product, String newProductId) {
+  // Update the product ID with the new value
+  product.productId = newProductId;
+}
+```
+
+- **Complex Property Modification - Nested Object Update**
+In this more complex example, we’ll modify a nested property within the `ProductStruct`, such as updating the price (which itself is a `PriceStruct` object). This shows how to update a property that itself contains multiple fields.
+
+```js
+// Function to update the price of a ProductStruct instance
+Future updateProductPrice(ProductStruct product, double newAmount, String currency) {
+// Check if price is not null
+    if (product.price != null) {
+        // Update only the amount field
+        product.price!.amount = newAmount;
+    } else {
+        // If price is null, optionally initialize it if needed
+        product.price = PriceStruct(
+            amount: newAmount,
+            currency: currency,
+    );
+    }
+}
+```
+
+- **Complex Property Modification - Updating a List Property**
+In this example, we’ll add new items to a list property, like adding new review comments to the `reviews` list in `ProductStruct`. This example shows how to work with a list of nested objects.
+
+```js
+Future addNewReviews(ProductStruct product) {
+  product.reviews ??= []; // Initialize the reviews list if it's null
+  product.reviews!.addAll([
+    ReviewStruct(rating: 5, comment: 'Excellent product!'),
+    ReviewStruct(rating: 4, comment: 'Good quality, but a bit expensive.'),
+    ReviewStruct(rating: 3, comment: 'Satisfactory, meets expectations.'),
+  ]);
+}
+```
+
+or if the new list of reviews are being provided to the Custom Action, then: 
```

Comment:
I think theres also a way to see if a field exists in a data type? I feel like that has come up. Might be a good example